WalletWasabi/Stores/SqliteStorage.cs

```diff
@@ -0,0 +1,241 @@
+using Microsoft.Data.Sqlite;
+using NBitcoin;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using WalletWasabi.Backend.Models;
+
+namespace WalletWasabi.Stores;
+
+/// <summary>
+/// Sqlite-based storage for block filters.
+/// </summary>
+/// <seealso href="https://learn.microsoft.com/en-gb/dotnet/standard/data/sqlite/async">
+/// Async is not supported at the moment, so no async methods were used in this implementation.
+/// </seealso>
+public class SqliteStorage : IDisposable
+{
+	private bool _disposedValue;
+
+	private SqliteStorage(SqliteConnection connection)
+	{
+		Connection = connection;
+	}
+
+	private SqliteConnection Connection { get; }
+
+	/// <summary>
+	/// Opens a new sqlite connection to the given database file.
+	/// </summary>
+	/// <param name="path">Path to the sqlite database file.</param>
+	/// <param name="startingFilter">Starting filter to put into the filter table if the table needs to be created.</param>
+	/// <exception cref="InvalidOperationException">If there is an unrecoverable error.</exception>
+	/// <seealso href="https://dev.to/lefebvre/speed-up-sqlite-with-write-ahead-logging-wal-do">Write-ahead logging explained.</seealso>
+	public static SqliteStorage FromFile(string path, FilterModel? startingFilter = null)
+	{
+		SqliteConnection connection = new($"Data Source={path}");
+		connection.Open();
+
+		using (SqliteCommand createCommand = connection.CreateCommand())
+		{
+			createCommand.CommandText = """
+				CREATE TABLE IF NOT EXISTS filter (
+					block_height INTEGER NOT NULL PRIMARY KEY,
+					block_hash BLOB NOT NULL,
+					filter_data BLOB NOT NULL,
+					previous_block_hash BLOB NOT NULL,
```

Comment:
Why BLOB not VARCHAR(64) for hashes?

Reply:
Ok, makes sense. Counter-argument would be that with varchars it's easier to examine database using sqlite tools. But this is anyway something that is not too hard to change in future if needed.

Reply:
> edit: But then I don't know much much actual data saving in kB/MB we are talking about from the top of my head.

64 bytes per block for `block_hash` and `previous_block_hash` combined. Is there all blocks or only after segwit activation? For all blocks it would be 781411*64=50010304, so around 50 MiB if my calculations are correct.